### ISSUE TITLE
in firefox, input elements of type "number" inhibit announcement of wrapped label

### ISSUE THREAD

**Issue by RichCaloggero** [nvaccess/nvda#5917]
Given code like:

```
<label>List length: <input type="number" min="1" max="10" value="5"></label>
```

NVDA only speaks value of control; label is not read.

Substituting other type (such as "range", "text", "tel", etc causes the label to be read just fine.

Latest version of Firefox and NVDA used.

**Comment by RichCaloggero** [comment]
Given this code:

```
<label>My number:
<input type="number"></label>
```

## Expected behavior

- nvda should say "My Number" when control gains focus
- up / down arrow keys should adjust value
- should be able to edit text of number as usual

## Current behavior

Note, as reiterated below, this only seems to happen on desktop machines.  Changing the keyboard setting in NVDA from desktop to laptop has no effect.


- label is not read when control gains focus
	+ label is read if label added via for attribute
- up / down arrows do change value, but in most cases, NVDA loses focus
- text of number can be edited as usual

## Operating system

windows 7, 64-bit

## Browser

Firefox 64-bit latest (54.0.1 at time of writing)


## Hardware

Dell desktop machine

__Note: when using laptop running windows 7, and all other factors equal, NVDA does allow adjusting value via up / down arrows without losing focus. I can think of no reason why this should be!!
However, when input type is "number", NVDA _never_ reads wrapped labels!

**Comment by jcsteh** [comment]
I can confirm that wrapped labels don't get read. This is a Firefox bug we'll need to file. (accName doesn't expose the label in this case.)

I can't reproduce the other issue. Can you elaborate on your statement concerning NVDA losing focus? What exactly happens? What happens if you press NVDA+tab after this occurs? Do you have "Automatic focus mode for caret movement" enabled or disabled?

**Comment by RichCaloggero** [comment]
When "Automatic focus mode for caret movement" is disabled, things behave as they should.

When I enable it, I get strange behavior Using this url:
`data:text/html,<input type="number" autofocus value="0">`

- control gains focus when url is loaded due to autofocus
- spin button is announced, followed by 0
- up arrow produces "spin button 0" while nvda+tab produces "edit focused 1"
- up arrow again produces "spin button 1" and nvda+tab produces "edit focused 2"
- down arrow produces "spin button 2" and nvda+tab produces "edit focused 1"


Adding the label as shown produces stranger behavior:
data:text/html,<label>My number: <input type="number" autofocus value="0"></label>

	- when control first gains focus, "spin button 0" is announced and nvda+tab produces "edit focused 0"
- when up arrow pressed, sound for leaving focus mode played and "spin button 0" announced, while nvda+tab says "edit focused 1"
- pressing down arrow now behaves as if control no longer has focus, i.e. NVDA reads label and control as a single utterence "my number: spin button 1"

**Comment by jcsteh** [comment]
Enabling automatic focus mode for caret movement means you want NVDA to
switch to browse mode when you hit the edge of an editable text area with
the up and down arrow keys. A spin button is still an editable text area,
so this behaviour does (and should) apply. Otherwise, pressing down arrow
to move to a spin button would get you into focus mode, but continuing to
press down arrow would never get you out of focus mode.

**Comment by jcsteh** [comment]
Filed [Mozilla bug 1473237](https://bugzilla.mozilla.org/show_bug.cgi?id=1473237).

**Comment by jcsteh** [comment]
Fixed in Firefox 74.
